Assets/Scripts/StateManager.cs

```diff
@@ -79,4 +229,13 @@ public class StateManager : MonoBehaviour {
             Debug.LogError("No object to delete");
         }
     }
+
+    public void ChangeMap()
+    {
+        currentSiteMarkers = new Dictionary<string, SiteMarker>();
+        currentLineMarkers = new List<LineMarker>();
+
+        earthSphere.SetActive(!earthSphere.activeSelf);
+        flatMap.SetActive(!flatMap.activeSelf);
+    }
```

Comment:
Shouldn't you also call `UpdateSites()` here, since you're clearing the currentSite/LineMarkers? Or is that already handled somewhere else?